### PR TITLE
[Qt3D] add missing qt port

### LINKED ISSUE #26908: [qt] qt3d libraries not available for Qt6
qt3d libraries (https://github.com/qt/qt3d) are not included in this port or as an independant port (ie qt3d) for Qt6, despite ports existing for qt5 (qt5-3d) and for Qt Quick (qtquick3d). Is there a reason for this?

### LINKED ISSUE #26986: [angle] not useable on linux
Try to link angle on linux results in:
```
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLESv2.a: undefined reference to `XNVCTRLQueryStringAttribute'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLESv2.a: undefined reference to `XNVCTRLIsNvScreen'
/usr/bin/ld: /mnt/vcpkg-ci/installed/x64-linux/debug/lib/libGLESv2.a: undefined reference to `XNVCTRLQueryExtension'
```

Missing:
https://github.com/google/angle/blob/2aa52da7e4c32ed6be32d25b487e2280c7b559da/BUILD.gn#L567-L576

Or to be more exact:
https://github.com/google/angle/blob/main/src/third_party/libXNVCtrl/BUILD.gn

blocks #26928

### PR DESCRIPTION
**Pull request by Neumann-A**
closes #26908 
closes #26986

~~blocked by #26986~~

### CODE REVIEW

File: ports/qt/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+      "features": [
+        "animation",
+        "assimp",
+        "extras",
+        "input",
+        "logic",
+        "qml",
+        "render",
+        "rhi",
+        "vulkan"
```
Comment (Neumann-A) [inline ports/qt/vcpkg.json:21]:
```suggestion
```

File: ports/qt/vcpkg.json [hunk 2]
```diff
@@ -1,10 +1,25 @@
...
+      "default-features": false,
+      "features": [
+        "animation",
+        "assimp",
+        "extras",
+        "input",
+        "logic",
+        "qml",
+        "render",
+        "rhi",
```
Comment (Neumann-A) [inline ports/qt/vcpkg.json:20]:
```suggestion
        "rhi"
```

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -885,6 +883,8 @@ qtwayland:x64-osx=fail
 qtwayland:arm64-osx=fail
 # Post build checks fail
 qtwebengine:arm64-windows=fail
+
+
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt:887]:
```suggestion
```

Comment (Neumann-A) [inline scripts/ci.baseline.txt:887]:
```suggestion
```

File: ports/egl-registry/portfile.cmake [hunk 4]
```diff
@@ -24,3 +24,5 @@ file(
   DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
   RENAME copyright
 )
+
+configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" "${CURRENT_PACKAGES_DIR}/share/egl/vcpkg-cmake-wrapper.cmake" @ONLY)
```
Comment (Neumann-A) [inline ports/egl-registry/portfile.cmake:28]:
```suggestion
configure_file("${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" "${CURRENT_PACKAGES_DIR}/share/egl/vcpkg-cmake-wrapper.cmake" @ONLY)

```

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (m-kuhn) [reply scripts/ci.baseline.txt:886]: Sidenote: github webinterface lets you suggest edits on multiple lines at once if you drag and drop the "plus button" for comments.

File: ports/egl-registry/vcpkg-cmake-wrapper.cmake [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${VCPKG_GLESV2_LIBRARY}")
+    #find_library(VCPKG_GL_LIBRARY NAMES GL) # FindOpenGL?
+    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${VCPKG_GL_LIBRARY}")
+    #find_library(VCPKG_ANGLE_LIBRARY NAMES ANGLE) 
+    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${VCPKG_ANGLE_LIBRARY}")
+    #find_library(VCPKG_XNVCNTRL NAMES XNVCtrl)
+    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${VCPKG_XNVCNTRL}")
+    #find_package(X11 COMPONENTS Xi Xext REQUIRED)
+    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "X11::X11;X11::Xext;X11::Xi")
+endif()
```
Comment (Neumann-A) [inline ports/egl-registry/vcpkg-cmake-wrapper.cmake:24]:
```suggestion
    #find_package(X11 COMPONENTS Xi Xext REQUIRED)
    #set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "X11::X11;X11::Xext;X11::Xi")
endif()
if(UNIX)
    find_library(VCPKG_XNVCNTRL NAMES XNVCtrl)
    set_property(TARGET EGL::EGL APPEND PROPERTY INTERFACE_LINK_LIBRARIES "${VCPKG_XNVCNTRL}")
endif()
```

File: scripts/ci.baseline.txt [hunk 6]
```diff
@@ -1257,6 +1261,12 @@ cmake-user:x64-windows-static-md=pass
 cmake-user:x64-windows-static=pass
 cmake-user:x64-windows=pass
 cmake-user:x86-windows=pass
+qt:x64-windows=pass
+qt:x64-windows-static=pass
+# qt:x64-windows-static-md=pass # hunspell is fail so this cannot pass.
```
Comment (Neumann-A) [inline scripts/ci.baseline.txt:1266]: Requires #26370 to move on. Hunspell uses MSBuild

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
@Neumann-A ,Thanks for your pr, when I tested the features by command "./vcpkg install qt3d[*]:x86-windows",
I get this error:
````
F:\Feature-test\qt3d\vcpkg\buildtrees\qtdeclarative\x64-windows-dbg\include\QtQuickDialogs2QuickImpl\6.3.2\QtQuickDialogs2QuickImpl/private/qquickfolderbreadcrumbbar_p.h(1): fatal error C1083: Cannot open include file: '../../../../../../src/here-src-6-3a791261d3.clean/src/quickdialogs2/quickdialogs2quickimpl/qquickfolderbreadcrumbbar_p.h': No such file or directory
````

**Comment by Neumann-A** [comment]
Buildtree too long... use --x-buildtree-root=<short-path>

**Comment by JonLiu1993** [comment]
All features are tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

**Pull request merged**